examples/Extensions/AddSitelinksUsingFeeds.cs

```diff
@@ -0,0 +1,401 @@
+            {
+                Create = feedMapping
+            };
+
+            MutateFeedMappingsResponse response = feedMappingServiceClient.MutateFeedMappings
+                (customerId.ToString(), new[] {operation});
+
+            Console.WriteLine($"Created feed mapping '{response.Results.First().ResourceName}'");
+        }
+
+        /// <summary>
+        /// Creates a campaign feed, which tells Google Ads which campaigns to use the provided
+        /// data with.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The customer ID for which the call is made.</param>
+        /// <param name="campaignId">The campaign to receive the feed.</param>
+        /// <param name="feed">The feed to connect to the campaign.</param>
+        private void CreateCampaignFeed(GoogleAdsClient client, long customerId, long campaignId,
+            Feed feed)
+        {
+            CampaignFeedServiceClient campaignFeedServiceClient =
+                client.GetService(Services.V3.CampaignFeedService);
+
+            // Fetch the Feed Item IDs and collapse them into a single comma-separated string.
+            List<long> feedItemIds = feed.Attributes.Select(attr => attr.Id.Value).ToList();
+            string aggregatedFeedItemIds = String.Join(',', feedItemIds);
+            CampaignFeed campaignFeed = new CampaignFeed()
+            {
+                Feed = feed.ResourceName,
+                Campaign = ResourceNames.Campaign(customerId, campaignId),
+                MatchingFunction = new MatchingFunction()
+                {
+                    FunctionString = $"AND(IN(FEED_ITEM_ID,{{ {aggregatedFeedItemIds} }})," +
+                                     "EQUALS(CONTEXT.DEVICE,'Mobile'))"
+                }
+            };
+            campaignFeed.PlaceholderTypes.Add(PlaceholderTypeEnum.Types.PlaceholderType.Sitelink);
+
+            CampaignFeedOperation operation = new CampaignFeedOperation()
+            {
+                Create = campaignFeed
+            };
+
+            MutateCampaignFeedsResponse response = campaignFeedServiceClient.MutateCampaignFeeds(
+                customerId.ToString(), new[] {operation});
+
+            Console.WriteLine($"Created campaign feed '{response.Results.First().ResourceName}'");
+        }
+
+        /// <summary>
+        /// Targets the feed items to the given ad group.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The customer ID for which the call is made.</param>
+        /// <param name="adGroupId">The ID of the Ad Group being targeted.</param>
+        /// <param name="feedItems">The feed items that were added to the feed.</param>
+        private void CreateAdGroupTargeting(GoogleAdsClient client, long customerId, long adGroupId,
+            List<string> feedItems)
```

Comment:
It is a bit misleading to pass an array of feeditems, and then only apply targeting to the first one in the list. Can we instead pass a single feeditem to this method, and move the comment about the "first feeditem" into the Run method where this method is being invoked?

Reply:
Makes sense to me, done.